assets/js/view/part/multi_revenue.rb

```diff
@@ -15,19 +15,30 @@ class MultiRevenue < Snabberb::Component
       }.freeze
 
       needs :revenues
-      needs :translate, default: 'translate(-30 0)'
+      needs :translate, default: 'translate(0 0)'
 
       def render
-        children = @revenues.flat_map.with_index do |rev, index|
+        # Compute text and width first in order to get total_width
+        computed_revenues = @revenues.map do |rev|
           phase, revenue = rev
           text = "#{'D' if phase == :diesel}#{revenue}"
 
-          color = phase == :diesel ? :gray : phase
-          fill = COLOR[color]
+          {
+            text: text,
+            width: text.size * 13,
+            color: phase == :diesel ? :gray : phase
+          }
+        end
 
-          width = text.size * 13
+        # Compute total width of rectangles so we can center
+        total_width = computed_revenues.inject(0) do |width, revenue|
```

Comment:
don't use inject, i think this is equivalent to reduce?

you can also just do copmute_revenues.sum { __ } instead

Reply:
Moving to `sum`, yeah `inject` and `reduce` are aliases :) I just thought pure ruby would want `inject` lol